### PR TITLE
Compilation of website fixes

### LINKED ISSUE #2769: [fujoshibitches] Connector not working
**Did you read [the troubleshooting guide](https://hakuneko.download/docs/troubleshoot/)**
**Yes**/No

**Is the website of the connector working properly / are you able to see the manga within your browser**
Yes/**No**

**Describe the bug**
"404 Not Found | openresty" error

**To Reproduce**
1. Name of connector visit http://fujoshibitches.shounen-ai.net/ 
2. Name of manga (if connector update is working)
3. Name of chapter (if manga update is working)
4. Copy any relevant error message that appears in a popup or (advanced) within the console (press F12)

**Screenshots**
If complex to describe, add screenshots to help explain your problem.
Screen Shot 2020-12-16 at 3 04 54 PM
Screen Shot 2020-12-16 at 3 05 03 PM
Screen Shot 2020-12-16 at 3 05 23 PM

**HakuNeko (please complete the following information):**
 - Version [6.17]
thank you

### LINKED ISSUE #2899: [Gekkou Scans] Outdated domain
**Did you read [the troubleshooting guide](https://hakuneko.download/docs/troubleshoot/)**
Yes

**Is the website of the connector working properly / are you able to see the manga within your browser**
No

**Describe the bug**
```diff
- http://leitor.mangascenter.com.br/
+ http://leitor.gekkouscans.com.br/
```

**HakuNeko (please complete the following information):**
 - Version 6.1.7

### LINKED ISSUE #2920: [loveheaven] Website changed to lovehug
**Did you read [the troubleshooting guide](https://hakuneko.download/docs/troubleshoot/)**
Yes

**Is the website of the connector working properly / are you able to see the manga within your browser**
Yes

**Describe the bug**
I think the website loveheaven.net changed to weloma.net since that is what it get's redirected to whenever I got to loveheaven

### LINKED ISSUE #2923: [mundowuxia] Connector not working
**Did you read [the troubleshooting guide](https://hakuneko.download/docs/troubleshoot/)**
Yes

**Is the website of the connector working properly / are you able to see the manga within your browser**
Yes

**Describe the bug**
https://mundowuxia.com/
The images are not downloaded properly, it has the following format ".bin" Example 012.bin

Whats your operating system?
	--> Windows 10 64 bits

Which desktop client version are you using?
	--> rev 6.17

What is the rev. number?
	--> @17a6fb
	
Which settings have you set in HakuNeko?
	--> Default

### LINKED ISSUE #3034: [MangaLib] Connector not working
**Did you read [the troubleshooting guide](https://hakuneko.download/docs/troubleshoot/)**
Yes

**Is the website of the connector working properly / are you able to see the manga within your browser**
Yes

**Describe the bug**
Сan't get chapter list from any manga

**To Reproduce**
1. MangaLib
2. Any
3. There is no chapters
4. BookmarkManager.mjs:137 Failed to load bookmarks: Invalid bookmark list!

**HakuNeko (please complete the following information):**
HakuNeko 8.3.4 @0c07e8
All setting was on default
Windows 10 19042.789

### LINKED ISSUE #3050: [Hikari Scan] Connector not working
**Did you read [the troubleshooting guide](https://hakuneko.download/docs/troubleshoot/)**
Yes

**Is the website of the connector working properly / are you able to see the manga within your browser**
Yes

**Describe the bug**
theme has been changed to MangaStream [?]

**Screenshots**
style_from_src
hakuneko_request

**HakuNeko:**
 - Version 6.1.7

### LINKED ISSUE #3051: [Mangas For You] Dead Scan
**Did you read [the troubleshooting guide](https://hakuneko.download/docs/troubleshoot/)**
Yes

**Is the website of the connector working properly / are you able to see the manga within your browser**
No

**Describe the bug**
"END OF AN ERA! WITH THIS I OFFICIALLY DECREE THE END OF THE SCAN! THANKS TO EVERYONE WHO READ AND SEE YOU SOON!" [...]
<https://www.facebook.com/mangasforyou/posts/3565802763477573>

**HakuNeko (please complete the following information):**
 - Version 6.1.7

### LINKED ISSUE #3053: [MangaJar] Connector not working
**Did you read [the troubleshooting guide](https://hakuneko.download/docs/troubleshoot/)**
Yes

**Is the website of the connector working properly / are you able to see the manga within your browser**
Yes

**Describe the bug**
Hakuneko only downloads the first page in each chapter

**To Reproduce**
1. Connector: MangaJar
2. Manga: F (Noboru Rokuda) https://mangajar.com/manga/f-noboru-rokuda/
3. Chapter: Download any chapter
4. Error: Failed to load resource: the server responded with a status of 404 (https://mangajar.com/manga/f-noboru-rokuda//chaptersList)

**HakuNeko (please complete the following information):**
 - Version 6.1.7

### LINKED ISSUE #3068: [rawsenmanga] Connector not working
**Did you read [the troubleshooting guide](https://hakuneko.download/docs/troubleshoot/)**
Yes

**Is the website of the connector working properly / are you able to see the manga within your browser**
Yes
**Describe the bug**
https://raw.senmanga.com/ broken. (no syncing, preview nor download)

### PR DESCRIPTION
**Pull request by 27136773**
Resolves #2769
Resolves #2899
Resolves #2920
Resolves #2923
References #2930
Resolves #3034
Resolves #3050
Resolves #3051
Resolves #3053
Resolves #3068

### CODE REVIEW

File: src/web/mjs/engine/Request.mjs [hunk 1]
```diff
@@ -478,6 +478,13 @@ export default class Request {
         }
         delete details.requestHeaders['x-sec-fetch-site'];
 
+        // HACK: Imgur does not support request with accept types containing other mimes then images
+        //       => overwrite accept header to prevent redirection to HTML notice
+        if(/i\.imgur\.com/i.test(uri.hostname)) {
```
Comment (Robonau) [inline src/web/mjs/engine/Request.mjs]: its not just imgur
many madara sites also need this

Comment (Robonau) [reply src/web/mjs/engine/Request.mjs]: see https://github.com/manga-download/hakuneko/issues/2925#issuecomment-760201931
Aloalivn hosting on Aloalivn responding with a web page and not an image 

edit: just tested with this PR, it breaks Aloalivn as well as the other connectors that https://github.com/manga-download/hakuneko/pull/2973 fixed

Comment (Robonau) [reply src/web/mjs/engine/Request.mjs]: note: the reason it had to be implemented anyways was a ["fix"](src/web/mjs/engine/Connector.mjs) for mangaku which broke after a couple days anyways

Comment (27136773) [reply src/web/mjs/engine/Request.mjs]: The previous `Accept` header was completely wrong. HakuNeko was telling each website for any request that it only accept images and no text, json, html, ...
It is a miracle that it lasted that long until MangaKu decided to respond correctly (in an attempt to block HakuNeko of course).

If there are more hosts that prefer to respond with HTML instead with the image when HakuNeko says "i accept html, text, json, images, ..." then detection for image links needs to be improved, so HakuNeko will say "i only accept images". As an alternative, affected connectors can still set the header in `_handleConnectorURI`.

Comment (Robonau) [reply src/web/mjs/engine/Request.mjs]: given _handleConnectorURI only deals with images, shouldn't the default behaviour be only accept images?

Comment (Robonau) [reply src/web/mjs/engine/Request.mjs]: the https://github.com/manga-download/hakuneko/pull/3082/commits/73efd906cb13f8857465c43725507713cb8ae75b commit seems to have fixed all known cases, but there is still edge cases where it wouldn't work and would require _handleConnectorURI to be altered

the images not having an image extension
but is probably edge enough to not worry about

File: src/web/mjs/connectors/GekkouScans.mjs [hunk 2]
```diff
@@ -1,13 +1,36 @@
...
-export default class GekkouScans extends FoolSlide {
+export default class GekkouScans extends MangaReaderCMS {
 
     constructor() {
         super();
         super.id = 'gekkouscans';
         super.label = 'Gekkou Scans';
         this.tags = [ 'manga', 'webtoon', 'portuguese' ];
-        this.url = 'http://leitor.mangascenter.com.br';
+        this.url = 'http://leitor.gekkouscans.com.br';
```
Comment (Sheepux) [inline src/web/mjs/connectors/GekkouScans.mjs]: Could switch to https

**Review by Sheepux — APPROVED** [review]
Looks Good To Me. Minor improvement possible

File: src/web/mjs/connectors/GekkouScans.mjs (hunk 2 above)
Comment (27136773) [reply src/web/mjs/connectors/GekkouScans.mjs]: Done

**Pull request merged**